src/test/java/unit/GcpDataprocClusterControlled.java

```diff
@@ -0,0 +1,244 @@
+package unit;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import bio.terra.cli.serialization.userfacing.resource.UFGcpDataprocCluster;
+import bio.terra.cli.serialization.userfacing.resource.UFGcsBucket;
+import bio.terra.cli.service.FeatureFlags;
+import bio.terra.cli.service.utils.CrlUtils;
+import bio.terra.cloudres.google.dataproc.ClusterName;
+import bio.terra.workspace.model.AccessScope;
+import harness.TestCommand;
+import harness.baseclasses.SingleWorkspaceUnitGcp;
+import harness.utils.GcpDataprocClusterUtils;
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+import org.hamcrest.CoreMatchers;
+import org.hamcrest.Matchers;
+import org.json.JSONObject;
+import org.junit.jupiter.api.Assumptions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests for the `terra resource` commands that handle controlled GCP clusters.
+ *
+ * <p>These tests share a single controlled cluster resource created before the tests run.
+ * Additional tests must be added before {@link #listDescribeReflectDelete()}
+ *
+ * <p>This test only executes on verily environments. Add a tag prefix to prevent it from running in
+ * the main repo
+ */
+@Tag("unit-verily-gcp")
+public class GcpDataprocClusterControlled extends SingleWorkspaceUnitGcp {
+
+  private UFGcpDataprocCluster testCluster;
+  private final String name =
+      "cliTestUserDataprocCluster" + UUID.randomUUID().toString().substring(0, 8);
+  private final String stagingBucketName =
```

Comment:
minor: use the same random suffix for all three fields

private final String randomUuid8 = UUID.randomUUID().toString().substring(0, 8);
private final String clusterName = "cliTestUserDataprocCluster" + randomUuid8;
private final String stagingBucketName = "staging-bucket-" + randomUuid8;
private final String tempBucketName = "temp-bucket-" + randomUuid8;